aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/mediation_denied_handler.py

```diff
@@ -0,0 +1,42 @@
+"""Handler for incoming mediation-deny-request messages."""
+
+from .....messaging.base_handler import (
+    BaseHandler,
+    BaseResponder,
+    HandlerException,
+    RequestContext,
+)
+
+from ..manager import MediationManager
+from ..messages.mediate_grant import MediationGrant
+from ..models.mediation_record import MediationRecord
+from aries_cloudagent.storage.error import StorageNotFoundError
+
+
+class MediationDenyHandler(BaseHandler):
+    """Handler for incoming mediation denied messages."""
+
+    async def handle(self, context: RequestContext, responder: BaseResponder):
+        """Message handler implementation."""
+        self._logger.debug(
+            "%s called with context %s", self.__class__.__name__, context
+        )
+        assert isinstance(context.message, MediationGrant)
+
+        if not context.connection_ready:
+            raise HandlerException("Invalid mediation request: no active connection")
+        try:
+            mgr = MediationManager(context)
+            _record = await MediationRecord.retrieve_by_id(
+                context, context.mediation_id
+            )
+            await mgr.deny_request(
+                mediation=_record,
+                mediator_terms=context.mediator_terms,
+                recipient_terms=context.recipient_terms
+            )
```

Comment:
`MediationManager.deny_request` is a server role method, not client. We should make the method names more explicit if necessary to clearly denote that. This usage of it here is incorrect.

Reply:
How is this incorrect use? deny_request updates state and create a denied message, while mediation_denied_handler.py does not use the created denied message, it does behave as desired. Should we separate the state update logic to be in the denied request handler? Maybe create new methods in the manager for updating the state?

Reply:
I think the real problem is the semantics of it. That sort of conceptual overlap will just be misleading and confusing, especially given that we already know that keeping straight what operations pertain to what role is difficult. This is a case where redundancy is preferable because of the further provided clarity.

Additionally, switching the state directly would take just as many lines as calling this method does anyways, fewer even! :P
```python
_record.state = MediationRecord.STATE_DENIED
await _record.save(context, reason='Client mediation request was denied')
```

Reply:
Astute observation, I will update the denied handler with the needed record update.